### ISSUE TITLE
Rubocop incorrectly complains about tabs in strings

### ISSUE THREAD

**Issue by drbrain** [bbatsov/rubocop#2446]
Create a file with a here-document.  If you copy-paste this example double-check it with `hexdump -C` or equivalent to make sure the tabs were copied out of github successfully (as preview mode appears to eat tabs).

```ruby
<<-'TAB_BUG'
	← tab
 	← space, tab
	 ← tab, space
TAB_BUG
```

Run rubocop against this file.

Expected results:

No offenses

Actual results:

Rubocop thinks it knows better than the string content author what content is allowed in a String.

```
$ ruby -Ilib bin/rubocop tabs_in_heredoc_bug.rb 
Inspecting 1 file
C

Offenses:

tabs_in_heredoc_bug.rb:1:1: C: Missing utf-8 encoding comment.
<<-'TAB_BUG'
^
tabs_in_heredoc_bug.rb:2:1: C: Tab detected.
	← tab
^
tabs_in_heredoc_bug.rb:3:2: C: Tab detected.
 	← space, tab
 ^
tabs_in_heredoc_bug.rb:4:1: C: Tab detected.
	 ← tab, space
^

1 file inspected, 4 offenses detected
```

**Comment by drbrain** [comment]
Oh, it fails for all strings unless:

```ruby
<<-'TAB_BUG'
	← tab
 	← space, tab
	 ← tab, space
TAB_BUG

'
	← tab
 	← space, tab
	 ← tab, space
'
```

```
Inspecting 1 file
W

Offenses:

tabs_in_strings_bug.rb:1:1: W: Literal <<-'TAB_BUG' used in void context.
<<-'TAB_BUG'
^^^^^^^^^^^^
tabs_in_strings_bug.rb:1:1: C: Missing utf-8 encoding comment.
<<-'TAB_BUG'
^
tabs_in_strings_bug.rb:2:1: C: Tab detected.
	← tab
^
tabs_in_strings_bug.rb:3:2: C: Tab detected.
 	← space, tab
 ^
tabs_in_strings_bug.rb:4:1: C: Tab detected.
	 ← tab, space
^
tabs_in_strings_bug.rb:8:1: C: Tab detected.
	← tab
^
tabs_in_strings_bug.rb:9:2: C: Tab detected.
 	← space, tab
 ^
tabs_in_strings_bug.rb:10:1: C: Tab detected.
	 ← tab, space
^

1 file inspected, 8 offenses detected
```

**Comment by jonas054** [comment]
I agree it's a bug. Needs to be fixed.

**Comment by alexdowad** [comment]
@drbrain, when reporting bugs against RuboCop, please run with `-D` so it shows the "cop" which is producing the erroneous warning (or correction).
